apps/api/src/modules/base_locale/base_locale.service.ts

```diff
@@ -193,12 +197,21 @@ export class BaseLocaleService {
   }
 
   async extractAndPopulate(baseLocale: BaseLocale): Promise<BaseLocale> {
+    const key = `${KEY_POPULATE_BAL_ID}#${baseLocale.id}`;
+    await this.cacheService.set(key, '');
     // On extrait la Bal de l'assemblage BAN ou la derniere revision sur l'api-depot
     const data: FromCsvType = await this.populateService.extract(
       baseLocale.commune,
     );
     // On populate la Bal
-    return this.populate(baseLocale, data);
+    const res = await this.populate(baseLocale, data);
+    await this.cacheService.del(key);
+    return res;
+  }
+
+  async isPopulating(baseLocale: BaseLocale): Promise<boolean> {
+    const key = `${KEY_POPULATE_BAL_ID}#${baseLocale.id}`;
+    return (await this.cacheService.get(key)) !== null;
```

Comment:
Nit : Je suis pas fan de ce genre de raccourcis, je trouve ça plus clair d'avoir 
const isPopulating = await this.cacheService.get(key)
return isPopulating !== null